### PR TITLE
Switch to installing GitVersion using dotnet tool install

### PR DESCRIPTION
**Pull request by johlju**
# Pull Request

### Changed
- Switch to installing GitVersion using `dotnet tool install`. Fixes [#348](https://github.com/gaelcolas/Sampler/issues/348)
- Updated pipeline to use the build worker image 'ubuntu-latest'.
- Updated pipeline to use the build worker image 'windows-latest'.
- Updated the Plaster templates
  - to use 'dotnet tool install' in the pipeline.
  - to use build image 'ubuntu-latest' in the pipeline.
  - to use build image 'windows-latest' in the pipeline.

## Task list

- [x] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [x] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [x] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [ ] Documentation added/updated in README.md.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/349)

### CODE REVIEW

**Comment by johlju** [comment]
Will send in more commits, just want to make sure the pipeline works.

File: Sampler/Templates/Sampler/azure-pipelines.yml.template [hunk 1]
```diff
@@ -40,7 +43,7 @@ stages:
               arguments: '-ResolveDependency -tasks pack'
               pwsh: true
             env:
-              ModuleVersion: $(gitVersion.NuGetVersionV2)
+              ModuleVersion: $$(NuGetVersionV2)
```
Comment (gaelcolas) [inline Sampler/Templates/Sampler/azure-pipelines.yml.template:46]:
```suggestion
              ModuleVersion: $(NuGetVersionV2)
```

Comment (gaelcolas) [reply Sampler/Templates/Sampler/azure-pipelines.yml.template:46]: Assuming that was a mistake.

**Review by gaelcolas — CHANGES_REQUESTED** [review]
One change and it's good to go.

**Review by gaelcolas — APPROVED** [review]
ok now, once the build passes.

**Pull request merged**